<launch>
    <arg name="AD0"       default="false"/>
    <arg name="PublishTF"       default="true"/>
    <arg name="UseCalibration"       default="true"/>
    <arg name="CalibrationTimes"       default="1000"/>
    <arg name="UseFilter"       default="true"/>
    <arg name="FilterGain"       default="0.031"/>
    <arg name="AccelerationScale"       default="0"/>
    <arg name="GyroScale"       default="0"/>
    <arg name="Hz"       default="10"/>
    <arg name="ParentsframeID"       default="base_link"/>
    <arg name="FrameID"       default="imu_link"/>
    <arg name="port"       default="/dev/ttyACM0"/>
	
    <node name="arduino_bringup" pkg="rosserial_arduino" type="serial_node.py" output="screen">
        <param name="port" value="$(arg port)" />
    </node>

    <rosparam file="$(find ydpibot_bringup)/param/pid.yaml" command="load" ns="pid"/>
    <rosparam file="$(find ydpibot_bringup)/param/robot.yaml" command="load" ns="robot"/>

    <node name="differential_drive_kinematic" pkg="ydpibot_bringup" type="diff_drive_kinematic.py" output="screen">
    </node>

    <include file="$(find ydlidar)/launch/lidar.launch"/>

  <node pkg="ydpibot_bringup" name="wheel_odom" type="odom.py"/>
  
  <node pkg="mpu6050" type="mpu6050_node" name="mpu6050_node" respawn="true">
    <param name="AD0" type="bool" value="$(arg AD0)" />
    <param name="PublishTF" type="bool" value="$(arg PublishTF)" />
    <param name="UseCalibration" type="bool" value="$(arg UseCalibration)" />
    <param name="CalibrationTimes" type="int" value="$(arg CalibrationTimes)" />
    <param name="UseFilter" type="bool" value="$(arg UseFilter)" />
    <param name="FilterGain" type="double" value="$(arg FilterGain)" />
    <param name="AccelerationScale" type="int" value="$(arg AccelerationScale)" />
    <param name="GyroScale" type="int" value="$(arg GyroScale)" />
    <param name="Hz" type="int" value="$(arg Hz)" />
    <param name="ParentsframeID" type="string" value="$(arg ParentsframeID)" />
    <param name="FrameID" type="string" value="$(arg FrameID)" />
  </node>
  
  <remap from="/imu_data" to="/imu/data"/>
  <remap from="/odom" to="/wheel_odom"/>
  <node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
    <param name="output_frame" value="odom_combined"/>
    <param name="base_footprint_frame" value="base"/>
    <param name="freq" value="10.0"/>
    <param name="sensor_timeout" value="1.0"/>
    <param name="odom_used" value="true"/>
    <param name="imu_used" value="true"/>
    <param name="vo_used" value="false"/>
    <param name="debug" value="false"/>
    <param name="self_diagnose" value="false"/>
  </node>

<param name="robot_description" command="cat '$(find ydpibot_description)/urdf/ydpibot_description.urdf'" />
  
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" >
  <param name="use_tf_static" value="false"/>
  </node>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" >
  </node>
  
</launch>